services/catalog/src/simcore_service_catalog/db/repositories/services.py

```diff
@@ -328,3 +341,105 @@ async def delete_service_access_rights(
+        """
+        logger.debug(
+            "getting specifications from db for %s", f"{key}:{version} for {groups=}"
+        )
+        gid_to_group_map = {group.gid: group for group in groups}
+        everyone_group_specs = None
+        team_group_specs: dict[GroupID, ServiceSpecificationsAtDB] = {}
+        user_group_specs = None
+
+        queried_version = packaging.version.parse(version)
+        # we should instead use semver enabled postgres [https://pgxn.org/dist/semver/doc/semver.html]
+        async with self.db_engine.connect() as conn:
+            async for row in await conn.stream(
+                sa.select([services_specifications]).where(
+                    (services_specifications.c.service_key == key)
+                    & (
+                        (services_specifications.c.service_version == version)
+                        if not allow_use_latest_service_version
+                        else True
+                    )
+                    & (
+                        services_specifications.c.gid.in_(
+                            (group.gid for group in groups)
+                        )
+                    )
+                ),
+            ):
+                try:
+                    logger.debug("found following %s", f"{row=}")
+                    # validate the specs first
+                    db_service_spec = ServiceSpecificationsAtDB.from_orm(row)
+                    db_spec_version = packaging.version.parse(
+                        db_service_spec.service_version
+                    )
+                    if allow_use_latest_service_version:
+                        # NOTE: in this case we look for the latest version only (e.g <=queried_version)
+                        # and we skip them if they are above
+                        if db_spec_version > queried_version:
+                            continue
+                    # filter by group type
+                    group = gid_to_group_map[row.gid]
+
+                    if group.group_type == GroupType.STANDARD:
+                        if _is_newer(
+                            team_group_specs.get(db_service_spec.gid), db_service_spec
+                        ):
+                            team_group_specs[db_service_spec.gid] = db_service_spec
+                    elif group.group_type == GroupType.EVERYONE:
+                        if _is_newer(everyone_group_specs, db_service_spec):
+                            everyone_group_specs = db_service_spec
+                    else:
+                        if _is_newer(user_group_specs, db_service_spec):
+                            user_group_specs = db_service_spec
+
+                except ValidationError as exc:
+                    logger.warning(
+                        "skipping service specifications for group '%s' as invalid: %s",
+                        f"{row.gid}",
+                        f"{exc}",
+                    )
```

Comment:
I guess this data is validated when inserted. What os the reason for skipping if a rule is not valid?
I would just let it rase an error in this case and fail.

Reply:
as a matter of fact, the data is NOT validated when inserted since it's an admin that will manually do it. therefore, I prefer validating before using. and in case it fails it just skips it, especially since there could be more than one specific spec